lucille-core/src/test/java/com/kmwllc/lucille/core/JsonDocumentTest.java

```diff
@@ -200,4 +202,26 @@ public void testJsonHandling() throws Exception {
     d.update("myJson", UpdateMode.APPEND, mapper.readTree("{\"a\":4}"));
     assertEquals(mapper.readTree("[{\"a\":1}, {\"a\":2}, {\"a\":3}, {\"a\":4}]"), d.getJson("myJson"));
   }
-}
\ No newline at end of file
+
+  @Test
+  public void testTransform() throws Exception {
+    Document doc = createDocumentFromJson("{\"id\":\"id\",\"foo\": \"bar\"}");
+    
+    // test mutating a reserved field
+    Expressions mutateReservedExpr = Expressions.parse("{\"id\":\"diff\",\"foo\": \"bar\"}");
+    assertThrows(DocumentException.class, () -> doc.transform(mutateReservedExpr));
+
+    // test mutatation does not create object 
+    Expressions mutateIntoArray = Expressions.parse("[1, 2, 3]");
+    Expressions mutateIntoNum = Expressions.parse("3");
+    assertThrows(DocumentException.class, () -> doc.transform(mutateIntoArray));
+    assertThrows(DocumentException.class, () -> doc.transform(mutateIntoNum));
+
+    // test valid mutation 
+    Expressions mutateFoo = Expressions.parse("{\"id\":\"id\",\"foo\": \"diff\"}");
+    doc.transform(mutateFoo);
+    assertEquals(2, doc.getFieldNames().size());
+    assertEquals("id", doc.getId());
+    assertEquals("diff", doc.getString("foo"));
+  }
```

Comment:
should include a test that applies an Expression that only operates on a specific field
this would be an expression that leaves all fields alone except one designated one
this expression shouldn't fully dictate the output json, it should just take the input (whatever it happens to be) and modifies it
our test should show that all the Document fields are intact after applying the expression, except the one we expect to be modified
it would be good to have one of the unmodified fields be a byte array field